7.0-cli/bin/magento-extension-installer

```diff
@@ -0,0 +1,424 @@
+            } else if ($opt_ind = array_search("--" . $option . "=" . $value, $arguments)) {
+                unset($arguments[$opt_ind]);
+            }
+        }
+
+        $arguments = array_values(array_filter(array_slice($arguments, 1), function ($value) {
+            return $value !== "--";
+        }));
+
+        if (count($arguments) > 1) {
+            throw new \Exception("Too many arguments!");
+        }
+
+        if (count($arguments) < 1) {
+            throw new \Exception("Must specify the name of the extension to install!");
+        }
+
+        $this->extensions_dir = implode(DIRECTORY_SEPARATOR, ["", static::EXTENSIONS_DIR]);
+
+        $this->extension_name = $arguments[0];
+        $this->source_dir = implode(DIRECTORY_SEPARATOR, [$this->extensions_dir, $this->extension_name]);
+
+        if (!$this->package_name) {
+            $this->package_name = $this->getComposerPackageName($this->source_dir);
+        }
+
+        $this->magento_dir = getenv("MAGENTO_ROOT");
+
+        if (!$this->magento_dir) {
+            throw new \Exception("This script requires the \$MAGENTO_ROOT environment variable to be defined!");
+        }
+
+        $this->symlink = implode(DIRECTORY_SEPARATOR, [$this->magento_dir, static::EXTENSIONS_DIR]);
+    }
+
+    /**
+     * Read the Composer package located at the given directory, verify that it
+     * is a Magento 2 extension and return the Composer package name.
+     *
+     * @param string $package_dir
+     *
+     * @return string
+     * @throws Exception
+     */
+    protected function getComposerPackageName($package_dir)
+    {
+        if (!file_exists($package_dir) || !is_dir($package_dir) || !is_readable($package_dir)) {
+            throw new \Exception(sprintf("Unable to read package directory %s!", $package_dir));
+        }
+
+        $manifest_file = $package_dir . DIRECTORY_SEPARATOR . "composer.json";
+
+        if (!file_exists($manifest_file) || !is_file($manifest_file) || !is_readable($manifest_file)) {
+            throw new \Exception(sprintf("Composer package not found in %s!", $package_dir));
+        }
+
+        $manifest = json_decode(file_get_contents($manifest_file), true);
+
+        if (!is_array($manifest) || !isset($manifest["type"]) || $manifest["type"] !== "magento2-module") {
+            throw new \Exception(sprintf("The Composer package in %s does not appear to be a Magento 2 module!", $package_dir));
+        }
+
+        if (!isset($manifest["name"]) || !$manifest["name"]) {
+            throw new \Exception(sprintf("Unable to determine the Composer package name in %s!", $package_dir));
+        }
+
+        return $manifest["name"];
+    }
+
+    /**
+     * Check if Magento 2 is installed in the given directory.
+     *
+     * @param string $directory
+     *
+     * @return bool
+     */
+    protected function isMagentoInstalled($directory)
+    {
+        $it = new \FilesystemIterator($directory);
+
+        return $it->valid();
```

Comment:
I guess a check for `app/etc/env.php` would prove that the installer was run?  Doesn't sound like it's necessary though if we're using it fine elsewhere.